pkg/repository/provider/unified_repo.go

```diff
@@ -185,6 +283,22 @@ func (urp *unifiedRepoProvider) getRepoOption(param RepoParam) (udmrepo.RepoOpti
 	return repoOption, nil
 }
 
+func (urp *unifiedRepoProvider) getRepoOptionForOpen(param RepoParam, purpose string) (udmrepo.RepoOptions, error) {
+	repoPassword, err := getRepoPassword(urp.credentialGetter.FromSecret, param)
+	if err != nil {
+		return udmrepo.RepoOptions{}, errors.Wrap(err, "error to get repo password")
+	}
+
+	repoOption := udmrepo.RepoOptions{
+		RepoPassword:   repoPassword,
+		ConfigFilePath: getRepoConfigFile(urp.workPath, string(param.BackupLocation.UID)),
+		Description:    purpose,
```

Comment:
It's a little confusing to set `purpose` to `Description`, is `purpose` a few constants that only kopia understands, or is it to control the flow, or is it just for logging?

By reading the code where `getRepoOptionForOpen` is called it looks like it's used to control the behavior of kopia, if that's the case it probably should not be `Description` in the option.

Reply:
The Description option defined by Unified Repo interface is not a option to control anything, it is basically for logging, for example, when opening a repository with the RepoOptions, we are able to know a description of, for example, the purpose of opening this repo, whether it is for backup, restore or maintenance?

This is not only an option that is understood by Kopia only, actually, anything we define in the Unified Repo interface should not be for Kopia only.

So here let me remove the parameter of "purpose" and use everything as "description", is it better?

Reply:
Thanks for the clarification.  I think it is better.  Could you also add some comment in the definition of the struct `RepoOptions` to clarify it should be used for logging only?

Reply:
Done, added the comment.